CMakeLists.txt

```diff
@@ -34,11 +34,14 @@ set(PICOQUIC_LIBRARY_FILES
     picoquic/spinbit.c
     picoquic/ticket_store.c
     picoquic/token_store.c
-    picoquic/tls_api.c
     picoquic/transport.c
     picoquic/util.c
 )
 
+set(PICOQUIC_CRYPTO_LIBRARY_FILES
+    picoquic/tls_api.c
+)
+
```

Comment:
Please don't do that. The glue to the picotls library really belongs in the core library.

I am not convinced that we need to handle the crypto implementations in a separate library, by opposition to keeping the files in the core library. Having the files in separate libraries just makes it harder for application. They will always need the crypto functions, so now they have to load two libraries instead of one.

Having a single library or two libraries has no effect on the size of the executable. Individual object files in the library only get loaded if the application uses functions or data from that file. If the crypto variant have different headers, the application will get to chose by inserting the chosen header and calling the "setup" function in the API.

See for example how "autoqlog" works.